cartoframes/client/data_obs_client.py

```diff
@@ -0,0 +1,692 @@
+            if suggested in table_columns:
+                names[suggested] = utils.unique_colname(suggested, table_columns)
+                warn(
+                    '{s0} was augmented as {s1} because of name '
+                    'collision'.format(s0=suggested, s1=names[suggested])
+                )
+            else:
+                names[suggested] = suggested
+
+        # drop description columns to lighten the query
+        meta_columns = _meta.columns.values
+        drop_columns = []
+        for meta_column in meta_columns:
+            if meta_column.endswith('_description'):
+                drop_columns.append(meta_column)
+
+        if len(drop_columns) > 0:
+            _meta.drop(drop_columns, axis=1, inplace=True)
+
+        cols = ', '.join(
+            '(data->{n}->>\'value\')::{pgtype} AS {col}'.format(
+                n=row[0],
+                pgtype=row[1]['numer_type'],
+                col=names[row[1]['suggested_name']])
+            for row in _meta.iterrows())
+        query = utils.minify_sql((
+            'SELECT {table_cols}, {cols}',
+            '  FROM OBS_GetData(',
+            '       (SELECT array_agg({how})',
+            '        FROM "{tablename}"),',
+            '       (SELECT \'{meta}\'::json)) as m,',
+            '       {tablename} as t',
+            ' WHERE t."{rowid}" = m.id',)).format(
+                how=('(the_geom, cartodb_id)::geomval'
+                     if how == 'the_geom' else how),
+                tablename=table_name,
+                rowid='cartodb_id' if how == 'the_geom' else how,
+                cols=cols,
+                table_cols=','.join('t.{}'.format(c) for c in table_columns),
+                meta=_meta.to_json(orient='records').replace('\'', '\'\''))
+
+        return self._fetch(query, decode_geom=False, table_name=persist_as)
+
+    def _fetch(self, query, decode_geom=False, table_name=None):
+        dataset = Dataset(query, credentials=self._old_context)
+        if table_name:
+            dataset.upload(table_name=table_name)
+        dataset.download(decode_geom=decode_geom)
+        return dataset
+
+    def _debug_print(self, **kwargs):
+        if self._verbose <= 0:
+            return
+
+        for key, value in utils.dict_items(kwargs):
+            if isinstance(value, requests.Response):
+                str_value = ("status_code: {status_code}, "
+                             "content: {content}").format(
+                                 status_code=value.status_code,
+                                 content=value.content)
+            else:
+                str_value = str(value)
+            if self._verbose < 2 and len(str_value) > 300:
+                str_value = '{}\n\n...\n\n{}'.format(str_value[:250],
+                                                     str_value[-50:])
+            print('{key}: {value}'.format(key=key,
+                                          value=str_value))
+
+    def _geom_type(self, source):
```

Comment:
A function very similar to this is already in data: https://github.com/CartoDB/cartoframes/blob/97c89b9ed8924275d98202a6ebf46a52c8903223/cartoframes/data/registry/base_dataset.py#L174